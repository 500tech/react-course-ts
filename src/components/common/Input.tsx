import * as React from 'react';

const Input = (props) => (
  <input
    ref={props.onRef}
    type="text"
    placeholder="something"
  />
);

export default Input;
