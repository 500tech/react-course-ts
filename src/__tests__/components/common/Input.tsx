import * as React from 'react';
import { shallow } from 'enzyme';

import { noop } from 'lodash';
import Input from "../../../components/common/Input";


describe('<Input />', () => {
  test('should render', () => {
    const component = shallow(
        <Input value="test" onChange={ noop } />
    );

    expect(component).toMatchSnapshot();
  });

  test('should call onChange when value changes', () => {
    const spy = jest.fn();
    const component = shallow(
        <Input value="test" onChange={ spy } />
    );

    expect(spy).not.toHaveBeenCalled();
    const event = { target: { value: 'John'} };
    component.simulate('change', event);

    expect(spy).toHaveBeenCalledWith(event);
  });
});


