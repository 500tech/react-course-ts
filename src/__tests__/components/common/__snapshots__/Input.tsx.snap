// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
<input
  onChange={[Function]}
  placeholder="new movie"
  type="text"
  value="test"
/>
`;
